src/client/components/pages/editor-revision.js

```diff
@@ -0,0 +1,150 @@
+		this.handleResultsPerPageChange = this.handleResultsPerPageChange.bind(this);
+	}
+
+	triggerSearch() {
+		const pagination = `size=${this.state.size}&from=${this.state.from}`;
+		request.get(`./revisions/revisions?${pagination}`)
+			.then((res) => JSON.parse(res.text))
+			.then((data) => {
+				this.setState(prevState => ({
+					nextEnabled: data.length >= prevState.size,
+					results: data
+				}));
+			});
+	}
+
+	handleClickPrevious() {
+		this.setState(prevState => ({from: Math.max(prevState.from - prevState.size, 0)}), this.triggerSearch);
+	}
+
+	handleClickNext() {
+		this.setState(prevState => ({from: prevState.from + prevState.size}), this.triggerSearch);
+	}
+
+	handleResultsPerPageChange(value) {
+		this.setState({size: parseInt(value, 10)}, this.triggerSearch);
+	}
+
+	render() {
+		return (
+			<div className="container">
+				<div id="EditorRevisionPage">
+
+					<RevisionsTable
+						results={this.state.results}
+						showRevisionEditor={this.props.showRevisionEditor}
+						showRevisionNote={this.props.showRevisionNote}
+						tableHeading={this.props.tableHeading}
+					/>
+
+					{
+						this.state.results && this.state.results.length ?
+							<div>
+								<hr className="thin"/>
+								<Pager>
+									<Pager.Item
+										previous disabled={this.state.from <= 0}
+										href="#" onClick={this.handleClickPrevious}
+									>
+										&larr; Previous Page
+									</Pager.Item>
+									<ButtonGroup>
+										<Button disabled>Results {this.state.from + 1} —
+											{this.state.results.length < this.state.size ?
+												this.state.from + this.state.results.length :
+												this.state.from + this.state.size
+											}
+										</Button>
+										<DropdownButton
+											dropup bsStyle="info" id="bg-nested-dropdown"
+											title={`${this.state.size} per page`}
+											onSelect={this.handleResultsPerPageChange}
+										>
+											<MenuItem eventKey="10">10 per page</MenuItem>
+											<MenuItem eventKey="20">20 per page</MenuItem>
+											<MenuItem eventKey="35">35 per page</MenuItem>
+											<MenuItem eventKey="50">50 per page</MenuItem>
+											<MenuItem eventKey="100">100 per page</MenuItem>
+										</DropdownButton>
+									</ButtonGroup>
+									<Pager.Item
+										next disabled={!this.state.nextEnabled}
+										href="#" onClick={this.handleClickNext}
+									>
+										Next Page &rarr;
+									</Pager.Item>
+								</Pager>
+
+							</div> :
+							null
+					}
+				</div>
+			</div>
+		);
+	}
+}
+
+
+EditorRevisionPage.displayName = 'EditorRevisionPage';
+EditorRevisionPage.propTypes = {
+	from: PropTypes.number,
+	results: PropTypes.array,
+	showRevisionEditor: PropTypes.bool,
+	showRevisionNote: PropTypes.bool,
+	size: PropTypes.number,
+	tableHeading: PropTypes.string
+};
+EditorRevisionPage.defaultProps = {
+	from: 0,
+	results: [],
+	showRevisionEditor: false,
+	showRevisionNote: false,
```

Comment:
Shouldn't the default be `true` here?

Reply:
If default is `true` then I have to pass `false` in props for the other thing and vice versa. I set it to `false` to avoid the case when editor or note is true but it isn't present 
I can change switch it if you want but I don't think it makes any difference.

Reply:
I'm not sure I understand what you mean by "I have to pass false in props for the other thing and vice versa".

I assume that specifically for this EditorRevisionPage, we always want to display the notes and don't want to display the editor, so the default props should reflect that, regardless of what is passed to the EditorRevisionPage 
 when it is instantiated.

Reply:
Oh I am so sorry I thought this was default props for revision table. My bad :(

Reply:
Yeah you're correct. I will have to change it in `revisionPage` also I guess. I will fix it. Thanks :)

Reply:
It's easy to get confused when looking at similar files for weeks :)
That's why pair programming and reviews are so useful — for everyone!